lib/msf/core/post/linux/system.rb

```diff
@@ -102,6 +102,216 @@ def get_sysinfo
+  end
+
+  #
+  # Gets the name of the current shell
+  # @return [String]
+  #
+  def get_shell_name
+    psout = cmd_exec('ps -p $$').to_s
+    psout.split("\n").last.split(' ')[3]
+  rescue
+    raise 'Unable to gather shell name'
+  end
+
+  #
+  # Checks if the system has gcc installed
+  # @return [Boolean]
+  #
+  def has_gcc?
+    command_exists? 'gcc'
+  rescue
+    raise 'Unable to check for gcc'
+  end
+
+  #
+  # Gets the version of gcc
+  # @return [String]
+  #
+  def gcc_version
+    return cmd_exec('gcc --version | head -n 1').to_s.split(' ')[3] if has_gcc?
+    raise 'System does not have gcc installed'
+  rescue
+    raise 'Unable to get gcc version'
+  end
+
+  #
+  # Checks if the `cmd` is installed on the system
+  # @return [Boolean]
+  #
+  def command_exists?(cmd)
+    cmd_exec("command -v #{cmd} && echo true").to_s.include? 'true'
+  rescue
+    raise "Unable to check if command `#{cmd}` exists"
+  end
+
+  #
+  # Gets the process id(s) of `program`
+  # @return [Array]
+  #
+  def pidof(program)
+    pids = []
+    full = cmd_exec('ps aux').to_s
+    full.split("\n").each do |pid|
+      pids << pid.split(' ')[1].to_i if pid.include? program
+    end
+    pids
+  end
+
+  #
+  # Checks if `file_path` is mounted on a noexec mount point
+  # @return [Boolean]
+  #
+  def noexec?(file_path)
+    mount = cmd_exec('cat /proc/mounts').to_s
+    mount_path = get_mount_path(file_path)
+    mount.lines.each do |l|
+      return true if l =~ Regexp.new("#{mount_path} (.*)noexec(.*)")
+    end
+    false
+  rescue
+    raise 'Unable to check for noexec volume'
+  end
+
+  #
+  # Checks if `file_path` is mounted on a nosuid mount point
+  # @return [Boolean]
+  #
+  def nosuid?(file_path)
+    mount = cmd_exec('cat /proc/mounts').to_s
+    mount_path = get_mount_path(file_path)
+    mount.lines.each do |l|
+      return true if l =~ Regexp.new("#{mount_path} (.*)nosuid(.*)")
+    end
+    false
+  rescue
+    raise 'Unable to check for nosuid volume'
+  end
+
+  #
+  # Checks for protected hardlinks on the system
+  # @return [Boolean]
+  #
+  def protected_hardlinks?
+    read_file('/proc/sys/fs/protected_hardlinks').to_s.strip.eql? '1'
+  rescue
+    raise 'Could not determine protected_hardlinks status'
+  end
+
+  #
+  # Checks for protected symlinks on the system
+  # @return [Boolean]
+  #
+  def protected_symlinks?
+    read_file('/proc/sys/fs/protected_symlinks').to_s.strip.eql? '1'
+  rescue
+    raise 'Could not determine protected_symlinks status'
+  end
+
+  #
+  # Gets the version of glibc
+  # @return [String]
+  #
+  def glibc_version
+    raise 'glibc is not installed' unless command_exists? 'ldd'
+    cmd_exec('ldd --version').scan(/^ldd\s+\(.*\)\s+([\d.]+)/).flatten.first
+  rescue
+    raise 'Could not determine glibc version'
+  end
+
+  #
+  # Gets the mount point of `filepath`
+  # @param [String] filepath The filepath to get the mount point
+  # @return [String]
+  #
+  def get_mount_path_of(filepath)
```

Comment:
This should be renamed `get_mount_path`, as per the `noexec?` and `nosuid?` methods.